### DISCUSSION TITLE
Welcome to radmc3d-2.0 Discussions!

### DISCUSSION THREAD

**Discussion by dullemond** [dullemond/radmc3d-2.0#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by astro-gioia** [comment]
Hi @dullemond and all, 

I am Gioia Rau, a research scientist at NASA/GSFC. I have been using the previous released version (V0.41) of RADMC3D to do some work on cool evolved stars. I have just seen the new release, but am considering to finish my work using V0.41, to avoid uninstalling the previous version and all the possibly related switching/compatibility issues. And then, of course, I will switch to V2.0 for future works.
What are the main changes to V2.0, except for added Python routines, for which I'd need to consider switching for concluding my current project? 
I've looked into the new V2.0 manual but I did not see a specific section on the changes; I'd appreciate if you please could address me to any places I can read more about this here on GitHub/on the manual, or if you please could directly comment on this.

Thank you very much in advance!
Best regards,

Gioia

**Reply by dullemond** [reply]
Dear Gioia,
Indeed, Version 2.0 is not fundamentally different from 0.41. The main differences are cleanups of stuff that was not used, a more self-consistent implementation of the Python packages, a cleaned up manual made with Sphinx, some improved opacity routines, some new example models. But the Fortran-90 code is still just adiabatically changing, no jump at the 0.41-2.00 boundary. So no worries :-). 
Best wishes,
Kees

**Reply by astro-gioia** [reply]
> Dear Gioia,
> Indeed, Version 2.0 is not fundamentally different from 0.41. The main differences are cleanups of stuff that was not used, a more self-consistent implementation of the Python packages, a cleaned up manual made with Sphinx, some improved opacity routines, some new example models. But the Fortran-90 code is still just adiabatically changing, no jump at the 0.41-2.00 boundary. So no worries :-).
> Best wishes,
> Kees

Dear Cornelis,
Great, thank you very much for your fast reply and clarifications! :)
Best regards,
Gioia

**Comment by anonm2f** [comment]
Hi

I am trying to simulate spectral lines of CO using the radmc3d code. I have been looking at the examples but there are some things I've not been able to understand.
What exactly is rhogas0. Is it the total gas density or H2 gas density? Is it in m^-2 or m^-3?
If I know the gas velocity of a cloud do i assign that single value to all the cells in the model?
Is the gas velocity being calculated in 'run_lines_lte_3d_1' example, the rotational velocity around the cloud centre or the galactic centre or something else?
If I have the characteristics of a cloud (dust temp, FWHM of observed lines, column density, etc.), how do I incorporate these in the code so as to get a line spectrum?
Also, can I incorporate the opacity effects due to ISM and earth's atmosphere?

Thanks

**Reply by dullemond** [reply]
Hi,
Whenever `rho` is written, it is density in g/cm^3 (Note: RADMC-3D uses CGS units!). And it is always _volume_ density (not surface density). As for the velocity: Yes, you have to assign that to all cells, but that also allows you to assign different velocities to different cells, for instance to include rotation. The velocity is (as always in physics) relative. If you want to add a velocity to the entire cloud you can easily do that _after_ you done the model calculation, by simply adding the systemic velocity to the x-axis.
As for how to incorporate things: Please try to figure out how the examples work, because the Python scripts in the `examples/` directory are meant for you to learn how this is done. RADMC-3D is not a "model", but a "code". The model is something you have to create yourself, starting with the examples in the `examples/` directory as templates.
Best wishes,
Kees

**Comment by tomek-kaminski** [comment]
Hi there,
I am not sure if this is the right place to do it but I found a few bugs in the current radmc3dPy procedures. 

1. In analyze. py, line 640, units km/h should be changed to km/s. This confused be a bit but it is an error in the procedure.

2. Using makeImage, imConv, and writeFits I am not able to get correct absolute values of flux, even though that plotting the image by plotImage gives correct values. I was not able to trace the scaling error. For now, I am exporting un-smoothed image to CASA and work there, but be aware there is a scaling error.

Has anyone here managed to produce spectral cubes in fits format that could be compared to ALMA observations?

**Reply by dullemond** [reply]
Hi Tomek,
Thanks for checking things so carefully. The unit must be indeed km/s. That appears to be a small typo. As for the imConv and writeFits: I don't know. The radmc3dPy package was not written by me, so it's a bit harder for me to quickly find any possible errors in that package. When you say you didn't get the correct absolute values of the flux, are you talking about errors of the order of percents or of factors of ten or so?
Cheers,
Kees

**Reply by tomek-kaminski** [reply]
Thanks, Kees, for looking at it. The problem with exported fits is that the values are many orders of magnitude off (conversion to Jy) but with some weird numeric factor I was not able to figure out in the script nor by tests on my model. 
Tomek

**Reply by Majiduzbak** [reply]
Hi tomek-kaminksi!
I am also trying to produce a fits file for the comparison to ALMA, My question is, did you figure it out how to? and also did you find out about the magnitude of the flux (which is in Jy/pixel) since it is much smaller than what we get from ALMA?

**Reply by secretgardeninmyword** [reply]
By setting plotImage(bunit='jy/pixel',dpc=1), the plotting image seems consists well with the generated fits file.

**Reply by astro-sophie** [reply]
Were you able to resolve this? I'm having similar issues.

**Comment by AlexisMatter** [comment]
Dear Kees,
i hope you are doing well. I've been using RADMC-3D for quite some time and have started using the latest 2.0 version.
When making images, i noticed that if i specify an odd number of pixels, the ray-tracing crashes (at least it did not work for me). The thing is that it worked perfectly fine in older versions. Is it normal ?
I'm mentioning that because with an even number of pixels, the star is never centered (since there is no central pixel) and always shifted from one pixel with respect to the circumstellar structure. That is a problem when computing interferometric phases out of the images because you always get a non-zeo interferometric phase even though your disk model is perfectly centro-symmetric.
Am i doing something wrong ?
Thanks for your help,
Alexis

**Reply by dullemond** [reply]
Hi Alexis,

No, you're doing nothing wrong. The problem with an odd number of pixels and perfect centering on the (0,0,0) point is that some formulae for the computation of the crossing of the rays with the cell ways become singular. I thought I had fixed that, but apparently it still causes problems. No idea why (in particular why it is not a problem with the older version). 

I see two quick-n-dirty fixes:

1. You can position the image on the sky completely the way you want, so you can move the centering of the image a minute bit off-center, too small to make a difference for the outcome, but avoiding the singularity.
2. You can by hand divide the star pixel over the 2x2 pixels at the center of the image. 

I know, neither are elegant, but at the moment I see no other quick solution. Note that you can also make images without the star ('nostar' command line option) and add the star by hand (unless you expect the star to be extincted somewhat). 

Does this help?

Best wishes,
Kees

**Reply by AlexisMatter** [reply]
Dear Kees,

Vielen Dank für die Antworte  :-)

I understand better the situation now. Concerning the two solutions you proposed :
1) If, in my image, there is anyway no central pixel and if the central star is on one single pixel, i don't see how to make it centered with respect to the circumstellar structure, whatever the position of the image on the sky (unless i make an image with a great number of pixels to reduce that relative offset). What matters is the relative position of the star with respect to the circumstellar structure. By the way, 
2) i think the second solution is probably the way to go. However, at which step do you think i should do it ? By hand, after making an image without the star included ? If so, that means that i should associate the star intensity to each of the 2*2 central pixels of the image ? In that case, the star would thus appear a bit 'spatially extended' in the image, which should not be a big problem if i consider a sufficiently small angular size for the pixels.

Thanks again,

Alexis

**Comment by pablorutschmann** [comment]
Hi @dullemond and all,

I have a problem with generating a function gird file for radmc3d. I succesfully created some images with `radmc3d image`but now i wanted to convert the grid to vtk format. If i run `radmc3d vtk_grid` i get the error message `ERROR: Inconsistent vertex: vidx = 1
icz=           1  icz0=       10881`. I have tried reformatting the `amr_grid.inp` file a few different ways, but didn't manage to get it working. How can the grid work for creating an image but not for the conversion? Also could you tell me what the `vidx`,`icz` and `icz0` are?

Thanks very much in advance.
Best Regards,
Pablo

**Reply by dullemond** [reply]
Dear Pablo,

hm, the VTK stuff is something I haven't used for quite a while. I am even unsure if it has been well-tested for non-Carstesian coordinates. I once created this to get a better 3-D understanding of 3-D models with AMR gridding. But these were all in cartesian coordinates. Do you use spherical coordinates?

Cheers,
Kees

**Reply by pablorutschmann** [reply]
Dear Kees,

yes i am using spherical coordinates. I now got it to work, there seemed to be a problem with the exactness of pi/2 and 2 pi. However i ran into another problem. The vtk command didnt seem to recognize binary input files. But it worked with ascii files.

Thanks for the response.

Best,
Pablo

**Comment by doug1r** [comment]
Dear Kees
I have been trying to run the lines_lte_0d example. But as I am increasing the sizex,y,z values from the default 100 au to a few pc the lines are becoming flat on the top and steeper on the wings, increasing the linewidth. Is this a physically correct result or a gridding problem for larger sizes
Thanks and Regards
Doug

**Reply by dullemond** [reply]
Yes, that is a well-known radiative transfer effect. See e.g.:

https://www.ita.uni-heidelberg.de/~dullemond/lectures/radtrans_2013/Chapter_3.pdf

**Comment by sachagavino** [comment]
Dear @Dullemond and everyone,

I am a postdoc at the University of Copenhagen and I have been using radmc3dPy for disks with multiple grain species.
I would like to notify (maybe) a potential bug when using mcmono.

I wrote a script that consecutively calls mctherm and mcmono and I noticed that when I set `scattering_mode_max` to any value greater than 1, the code fails when calling 'mcmono', with the following error:

`Note: The following floating-point exceptions are signalling: IEEE_DENORMAL
STOP 4097`

The value must be switched to 1 (or 0) in order to carry on and compute the local radiation field.

I may just do something wrong, but since scattering is a central parameter for the study I am working on, I would be happy to know why this is happening, and if there is anything I should do better.

Thank you and have a great day!

Sacha

**Reply by dullemond** [reply]
Hi Sacha,
Seems to be a mode-incompatibility. If you have an axisymmetric coordinate system (2D spherical coordinates r, theta, but not phi), then scattering is only possible (for technical reasons) for isotropic scattering or for the full-fledged scattering with the scattering matrix (scattering mode 5), nothing in between. This is because for non-isotropic scattering, the model must be "expanded" internally to a 3D form, because the scattering source function then depends on angle phi (even if all the rest of the model does not depend on phi). This "expansion" is a relatively new capability of RADMC-3D, and I never bothered to build it in for all scattering modes. I only did so for scattering mode 5 (the most complete one). 
But I see that you are "only" interested in computing the mean intensity, not the full scattering source function. In that case it should not be neccessary to exclude scattering modes 2 to 4, but apparently it is excluded still. Sorry for this. A work-around is if you simply use the full scattering matrix (scattering mode 5).
Best wishes,
Kees

**Reply by sachagavino** [reply]
Thanks for the prompt reply!
I understand, that makes sense as I restricted my models to 2D spherical coordinates and I never tried to push the mode to 5. But working with the full scattering matrix won't hurt, so I guess the problem is solved :)
Thanks again.
Best regards,
Sacha

**Comment by jwiegert** [comment]
Hi Dullemond (and all that are interested),

I'm experimenting with Radmc3d on circumstellar AGB star dust and trying to solve a few shortcuts and approximation I've had to use in my previous work.

The model I'm considering to create would require quite a number of different dust species. Initially due to several grain sizes in different spatial regions around the star (some 10 to 20 per chemical composition [probably 2-3 of these] I believe right now), plus an as-of-yet unknown amount due to the star that I'm trying to implement. That is, I'm experimenting with translating a simulated inhomogeneous luminous cool giant star into a ball of "dust" that recreates the spectral behaviour and temperatures of the star. Temperatures are thus already given and won't be simulated in Radmc3d.

- How many dust species can Radmc3d handle before it crashes? 
- OR should I perhaps modify (or is it even possible to modify) the source code for Radmc3d so that it re-computes the kappa_abs and kappa_scat for each spatial grid cell (using the included BH-code) based on new input data containing grain sizes and optical properties for each spatial grid?

I'm thinking that this second alternative would be quite computational heavy, especially for >1e7 grid cells and that just having a lot of -up to 100- pre-made dust species would be more effective. But knowing if there are any limits other than the amount of memory on my cluster would be good.

- Is there a way to include an inhomogeneous star as Radmc3d is now?

Thanks and cheers!
Joachim

**Reply by dullemond** [reply]
Dear Joachim,

RADMC-3D has no intrinsic limit on the number of dust species. But of course, at some point you run into memory limits. 

You can modify the code whereever/however you like. I'm sure it should be doable to call a function that returns the local kappa_nu in the specific grid cell. It might make the code very slow, though (if you want to use the Monte Carlo module). And if you store the kappa_nu at each cell, you would run into memory problems again. 

But if you fix the temperatures anyway (at least for some of the dust species), then those "fixed temperature dust species" could be implemented as a function returning kappa_nu. 

It is all possible, but it would require you to make a major modification to the code. 

I think that I once envisaged a gas emission/absorption part. Some remnants are still in the code (see `gascontinuum_module.f90`), but it has never been put into practice. I am not sure if this works or if it is any good. But it does appears to be the kind of thing you may need. Have a look at:

 - `gascontinuum_module.f90/gascont_addto_jnu_alpnu()`
 - `sources_module.f90` line 1135 where `gascont_addto_jnu_alpnu()` is called.

If you get it to work, then I'd love to hear about it! 

Best wishes,
Kees

**Reply by jwiegert** [reply]
Thanks for your fast response!

It seems my initial suspicions are confirmed. I will probably go on with using a large number of dust species for my models since calculating the kappa's for each cell would be slow. Also, it will be easier to implement and I'll only need to create my dustkappa_*-files once for all models.

My collaborators (who know fortran, which I don't :) ) may be interested in looking into the unused gas continuum modules. I agree that it could be of interest for this project. I promise to write here if we are able to make some progress with them!

Thanks again, and have a nice weekend,
Joachim

**Comment by AlexisMatter** [comment]
Dear Kees,
I hope you are doing well.
I come back to you concerning a strange error my student got when creating image cubes using either the makeimage routine from radmc3dPy or simply the command 'radmc3d image lambdarange 3.0 4.0 nlam 5'. Indeed, as he is trying to compute the images, radmc keeps stopping before the first Monte Carlo run and displaying 'ERROR: You try to use two methods for setting the camera wavelengths.'
This is really strange since when running exactly the same code with the same RADMC-3D version (the latest one) on my side, I get zero error and everything is fine. Even after reinstalling the entire code, he got the same error.
We also tried to start from a minimalistic directory with only the script and one opacity file and he gets the same error although obviously he doesn't use two methods to set the camera wavelength. Interestingly, the image cube computation works fine when using the loadlambda option or the allwl option. Only when he uses the lambdarange thing, it crashes. 
I checked the camera_module.f90 routine, since the error message comes from it, but could not find anything useful.
Have you ever encountered such an error when computing image cubes?
Thank you again for your help,
ein schoenen Abend.

Alexis

**Reply by dullemond** [reply]
Hi Alexis,
you are right. This is a bug. I fixed it. Please check if the bug is indeed fixed in the latest commit (ecde841).
Cheers,
Kees

**Reply by AlexisMatter** [reply]
Dear Kees,

thank you very much for your quick reply and fix.
My student installed the newest version and it works fine now. No error message anymore when using the lambdarange option.
Vielen Dank,
Alexis

**Comment by vtcloud** [comment]
Dear Kees,
thank you very much for making radmc3d publicly available and updating it. Currently I am running simple CO slab models to explore parameter spaces for observations. There are 2 issues I'm trying to understand:
1. The code allows me to insert turbulence into the model. From looking at the code, I get that the turbulence velocities are v_sigma. Since I am coming from v_FWHM, I have to divide by the usual sqrt(8 ln2) factor. However, when I am comparing my results with results from a simple analytical model (e.g., Myers et al 1993), I am off by a factor of by a factor of sqrt(2) looking at the FWHM line width of the Myers model compared to the radmc3d model. The equation in the radmc3d manual in Ch 7.4.8 shows the factor 2 in the part for the thermal broadening. Is this the possible origin of the missing factor sqrt(2) I have?
2. I am trying to convert the resulting spectrum from Jy to the more familiar brightness temperature using Tb = (lambda^2*Omega)/(2 k) * Snu. However, the brightness temperatures I'm getting out are too small using an Omega for a 1.2-m telescope with theta_beam=8.4 arcmin. Do you have any suggestion I should look into here?
Thank you very much for your help,
Volker

**Reply by dullemond** [reply]
Hi Volker,
1. See manual for the definition of the thermal line width. I think that's where the factor 2 is.
2. Not sure about the equation. Use the Planck function (the version in frequency B_nu, _not_ the version in wavelength B_lambda!), either the full one or the linearlized (Rayleigh-Jeans) one.
Cheers,
Kees

**Reply by vtcloud** [reply]
Hi Kees,
thank you for your response. I found the answer to my first question just now in your lecture notes (Chapters 7.51 and 7.52). The problem was between 1d velocities and 3d velocities. I will work on my second question later.
Cheers,
Volker

**Comment by mzainmob** [comment]
Hi @dullemond, I am currently trying to understand how to implement polarization when making images with RADMC3D. In particular, I would like to incorporate linear polarization into the models for our object and make images in polarized emission. I am not entirely sure as to how to go about this. Any advice on this matter would be extremely helpful. Thanks!

Regards,
Zain

**Comment by sachagavino** [comment]
Hello @dullemond,

I have two questions.
1) I am wondering how the variable 'itempdecoup' works when it is set to 0 with N dust species. Does the resulting temperature follows an area-weighted temperature of the N species, or something like that?

2) How is scattering treated for dust species with a very low density or density at zero? Intuitively, I would think that there should not be scattered emission coming from dust species with a density at zero. However, in a model with N dust species, I noticed after doing tests that the scattering from a dust species with a density set to zero everywhere in the model still affects the other coexisting dust species (their temperature, in particular). But maybe I am missing something here. 

Thanks a lot.
Cheers,
Sacha

**Reply by dullemond** [reply]
Hi Sacha,

1. It is the emission-weighted temperature. The easiest way to think of it is to assume that there is just 1 grain, with the opacities given by the sum_i rho_i kappa_{i,nu}.
2. That is weird. If the density is 0 it should not affect the temperature. Could you send me a tiny model setup (which should ideally finish running in at most a few minutes, please) where I can see what happens? That way I can do trouble shooting.

Best wishes,
Kees

**Comment by tomek-kaminski** [comment]
Hi LKV791,
Unfortunately, I could not fix the issue. The display in python/ipython
shows units correctly and only on export to fits something gets scrambled.
I used the display to calculate - at a given pixel  -- what was the
conversion factor and then scaled the entire fits file data section.
I am sure someone has fixed it and would be great if they shared the bug
fix with us.
Regards, Tomek


On Tue, Jan 30, 2024 at 2:59 PM LKV791 ***@***.***> wrote:

> Hi tomek-kaminksi!
> I am also trying to produce a fits file for the comparison to ALMA, My
> question is, did you figure it out how to? and also did you find out about
> the magnitude of the flux (which is in Jy/pixel) since it is much smaller
> than what we get from ALMA?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dullemond/radmc3d-2.0/discussions/3#discussioncomment-8305893>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATHXQAGKKOD2OU57R7HMYUTYRD4DRAVCNFSM4VVGN2B2U5DIOJSWCZC7NNSXTOKENFZWG5LTONUW63SDN5WW2ZLOOQ5TQMZQGU4DSMY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*        Tomasz (Tomek) Kamiński*
                      * * * * *
                   professor at
    Laboratory for Astrophysics in Toruń
*Nicolaus Copernicus Astronomical Center*
*   of the Polish Academy of Sciences,*
     Rabiańska 8, 87100 Toruń, Poland

**Reply by AlexisMatter** [reply]
Hi Tomek,
I used many times the writeFits routine and did not encounter any problem as far as I can remember. Did you always insure to include the right distance (in pc) of your object for the conversion in Jy/pix in writeFits? 
Were the images you put into WriteFits still in intensity/pix (i.e. in erg s−1 cm−2 Hz−1 ster−1)? I am asking you that because the imConv routine is not supposed to change the brightness unit of the input image and should thus provide you with a convolved image in intensity/pix.
I'm gonna investigate on my side,
Alexis

**Comment by tomek-kaminski** [comment]
Thanks, Alexis. I forgot the details, but imConv was involved and the
issues is with Jy/beam which is not Jy/pix.
Tomek

On Wed, Jan 31, 2024 at 9:21 AM AlexisMatter ***@***.***>
wrote:

> Hi Tomek,
> I used many times the writeFits routine and did not encounter any problem
> as far as I can remember. Did you always insure to include the right
> distance (in pc) of your object for the conversion in Jy/pix in writeFits?
> Were the images you put into WriteFits still in intensity/pix (i.e. in erg
> s−1 cm−2 Hz−1 ster−1)? I am asking you that because the imConv routine is
> not supposed to change the brightness unit of the input image and should
> thus provide you with a convolved image in intensity/pix.
> I'm gonna investigate on my side,
> Alexis
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dullemond/radmc3d-2.0/discussions/3#discussioncomment-8316286>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATHXQAAQHX3K5C4U236LCYDYRH5IXAVCNFSM4VVGN2B2U5DIOJSWCZC7NNSXTOKENFZWG5LTONUW63SDN5WW2ZLOOQ5TQMZRGYZDQNQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*        Tomasz (Tomek) Kamiński*
                      * * * * *
                   professor at
    Laboratory for Astrophysics in Toruń
*Nicolaus Copernicus Astronomical Center*
*   of the Polish Academy of Sciences,*
     Rabiańska 8, 87100 Toruń, Poland

**Comment by ciskakemper** [comment]
Hi, Kees, all,

My student Maryam and I are new users of RADMC-3D. We are trying to follow the installation instruction on the installation of the python tools, and while everything seems to go well, and we are setting the path and all that correctly, we continue to find that the python3 does not find natconst for some reason. It is finding simpleread without any issues. Here is the trace. What could be wrong?

`
>>> import matplotlib.pyplot as plt
>>> from radmc3d_tools.simpleread import *
>>> import radmc3d_tools.natconst as nc
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/maryam/bin/python/radmc3d_tools/__init__.py", line 8, in <module>
    from . import sph_to_sphergrid
  File "/home/maryam/bin/python/radmc3d_tools/sph_to_sphergrid.py", line 2, in <module>
    from natconst import *
ModuleNotFoundError: No module named 'natconst'
`
We don't know how to continue. For your information, we are using python3, in a virtual environment, and $HOME/bin/python has been added to the python path.

Thanks! 
Ciska Kemper & Maryam Torki

**Reply by ciskakemper** [reply]
OK, it turns out that adding $HOME/bin/python/radmc3d_tools to the $PYTHONPATH env variable actually does the trick, and the code now finds natconst. So that was not stated correctly in the manual. However, we are finding a new problem and that is that np.int in simpleread.py is deprecated. It won't run. The error message suggests to use int instead of np.int, but even with that, it still does not run. Any thoughts? 
Thanks!

**Comment by tomek-kaminski** [comment]
No, Sophie, I tried, but could not fix it in the code. I scale fits images
manually for each project, trusting that the python display is correct.
Tomek

On Mon, Jul 22, 2024 at 9:34 PM Sophie Clark ***@***.***>
wrote:

> Were you able to resolve this? I'm having similar issues.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dullemond/radmc3d-2.0/discussions/3#discussioncomment-10118752>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATHXQAFI67WIWGMTLLD3FUTZNVNCHAVCNFSM6AAAAABKSR33EGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJRHA3TKMQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*        Tomasz (Tomek) Kamiński*
                      * * * * *
                   professor at
    Laboratory for Astrophysics in Toruń
*Nicolaus Copernicus Astronomical Center*
*   of the Polish Academy of Sciences,*
     Rabiańska 8, 87100 Toruń, Poland

**Comment by ShivaThanay007** [comment]
Hello everyone!

I'm an undergraduate student and I've been studying post-AGB stars. I'm new to RADMC-3D, and I wanted to build models of the circumstellar environment of these stars to get their spectra. However, I don't particularly have a lot of information about the stars other than their luminosity. Is it possible for me to make models with multiple parameter values at once, and get the best-fit model or something of that sort?

Thanks!
